query-node/schema.graphql

```diff
@@ -65,55 +52,40 @@ type ChannelCategory @entity {
 
   channels: [Channel!] @derivedFrom(field: "category")
 
-  happenedIn: Block!
+  happenedIn: BigInt!
 }
 
-"Storage asset"
-union Asset = AssetUrl | AssetStorage
+"Asset availability representation"
+enum AssetAvailability {
+  ACCEPTED, # asset is available in storage
+  PENDING, # asset is being uploaded to storage
+  URL, # asset is referencing URL
+  INVALID, # invalid storage (meta)data used
+}
+
+"Asset representation"
+union Asset = AssetUrl | AssetStorage | AssetDeleted
 
 "Asset stored at an external source"
 type AssetUrl @variant {
-  id: ID!
-
   "The http url pointing to the media"
   url: String!
 }
 
-"Asset was never fully uploaded."
-type AssetNeverProvided @variant {
-  happenedIn: Block!
-}
-
 "Asset was deleted and is no longer available."
 type AssetDeleted @variant {
-  happenedIn: Block!
+  happenedIn: BigInt!
 }
 
-"Status of an asset upload"
-type AssetUploadStatus @variant {
+"Storage asset"
+type AssetStorage @variant {
```

Comment:
Do we actually need this now? Couldn't we link directly to `AssetDataObject`? `AssetStorage` has an additional `happenedIn` but `AssetDataObject` has `addedAt` so I don't think we're losing any data?

Reply:
No, we don't.  Now the whole structure is simplified and flatten.